app/controllers/donations_controller.rb

```diff
@@ -0,0 +1,27 @@
+class DonationsController < ApplicationController
+
+  helper_method :resource
+
+  def create
+    raise Pundit::NotAuthorizedError if !current_user
+    @donation = Donation.create
+    @donation.notify(:contribution_donated, current_user)
+    update_pending_refunds
+  end
+
+  def resource
+    @donation
+  end
+
+  private
+  def update_pending_refunds
+    current_user.pending_refund_payments.each do |payment|
+      payment.contribution.update_attribute :donation, @donation
+
+      payment.state = 'refunded'
+      payment.save!
```

Comment:
@diogob @vicnicius @adrianob we need to create a badge on contributions admin to show to support team that this these contributions is donations, because when they look at payment gateway dashboard the state is 'paid'